statistics/scalar.go

```diff
@@ -215,7 +215,7 @@ func enumRangeValues(low, high types.Datum, lowExclude, highExclude bool) []type
 			return nil
 		}
 		remaining = remaining + 1 - uint64(exclude)
-		if remaining >= maxNumStep {
+		if remaining >= maxNumStep || remaining < 0 {
```

Comment:
Seems like we only need to check time-related types because other types cannot cause overflow? The PR is partially overlapped with https://github.com/pingcap/tidb/pull/23890.